### PR TITLE
Add async franklin algorithm implementation

### PR DESCRIPTION
**Pull request by tymoteusz-chelmecki**

### CODE REVIEW

File: leader/undirected_ring/async_franklin/async_franklin.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+		})
+		node.SendMessage(LEFT, byteMsg)
+		node.SendMessage(RIGHT, byteMsg)
+	} else {
+		log.Println("[LEADER] Node", node.GetIndex(), "becomes leader")
+		setFinalState(node, id, true)
+	}
+}
+
+func handleTwoMessages(node lib.Node, first int, second int, firstMsgBytes []byte) (uint64, uint64) {
```
Comment (krzysztof-turowski) [inline leader/undirected_ring/async_franklin/async_franklin.go:103]: first parameter is not used, please consider refactoring the function

File: leader/undirected_ring/async_franklin/async_franklin.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+func handlePassive(node lib.Node) {
+	var sender int
+	var byteMsg []byte
+	var msg messageContent
+
+	log.Println("Node", node.GetIndex(), "becomes passive")
+	for {
+		sender, byteMsg = node.ReceiveAnyMessage()
+		if sender == LEFT {
```
Comment (krzysztof-turowski) [inline leader/undirected_ring/async_franklin/async_franklin.go:121]: Please consider:
other := RIGHT
if sender == RIGHT {
  other = LEFT
}
decodeAll(byteMsg, &msg)
node.SendMessage(other, byteMsg)
if msg.IsFinal {
  setFinalState(node, msg.Value, false)
  return
}

**Review by krzysztof-turowski — CHANGES_REQUESTED** [review]
Thank you very much - just small changes in the code needed, would you be finishing it or should I do it on my own?

**Review by krzysztof-turowski — COMMENTED** [review]
Just one more thing: please add appropriate info to Makefile and README

**Comment by tymoteusz-chelmecki** [comment]
I've added requested changes in a new commit.

**Review by krzysztof-turowski — APPROVED** [review]
Thank you very much for this nice contribution

**Pull request merged**